src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java

```diff
@@ -479,6 +482,82 @@ public void deleteSiteDependencies(@ValidateStringParam(name = "site") String si
         dependencyMapper.deleteDependenciesForSite(params);
     }
 
+    @Override
+    public Set<String> getDeleteDepenencies(String site, String path) throws SiteNotFoundException, ContentNotFoundException, ServiceException {
+        List<String> paths = new ArrayList<String>();
+        paths.add(path);
+        Set<String> processedPaths = new HashSet<String>();
+        return getDeleteDependenciesInternal(site, paths, processedPaths);
+    }
+
+    @Override
+    public Set<String> getDeleteDepenencies(String site, List<String> paths) throws SiteNotFoundException, ContentNotFoundException, ServiceException {
+        Set<String> processedPaths = new HashSet<String>();
+        return getDeleteDependenciesInternal(site, paths, processedPaths);
+    }
+
+    private Set<String> getDeleteDependenciesInternal(String site, List<String> paths, Set<String> processedPaths) {
+        Set<String> toRet = new HashSet<String>();
+        // Step 1: collect all content from subtree
+        Set<String> children = getAllChildrenRecursively(site, paths);
+        toRet.addAll(children);
+
+        // Step 2: collect all dependencies from DB
+        Set<String> dependencies = getItemDeleteDependencies(site, paths);
+        toRet.addAll(dependencies);
+        boolean doItAgain = false;
+
+        doItAgain = doItAgain || processedPaths.addAll(children);
+        doItAgain = doItAgain || processedPaths.addAll(dependencies);
+        if (doItAgain) {
+            List<String> pathsToProces = new ArrayList<String>();
+            pathsToProces.addAll(children);
+            pathsToProces.addAll(dependencies);
+            toRet.addAll(getDeleteDependenciesInternal(site, pathsToProces, processedPaths));
+        }
+
+        // Step 4: recursion ???
+        return toRet;
+    }
+
+    private Set<String> getItemDeleteDependencies(String site, List<String> paths) {
+        Map<String, Object> params = new HashMap<String, Object>();
+        Set<String> toRet = new HashSet<String>();
+        boolean exitCondition;
+        do {
+            params = new HashMap<String, Object>();
+            params.put("site", site);
+            params.put("paths", paths);
+            List<String> deps = dependencyMapper.getDependenciesForList(params);
+            Set<String> filtered = new HashSet<String>();
+            filtered.addAll(deps);
+            filtered = filterItemSpecificDependencies(filtered);
+            exitCondition = !toRet.addAll(filtered);
+            paths.clear();
+            paths.addAll(deps);
+        } while (!exitCondition);
+        return toRet;
+    }
+
+    private Set<String> getAllChildrenRecursively(String site, List<String> paths) {
```

Comment:
Doesn't this method exist somewhere else as some form of getTree? Or is this here because it returns a simple list of paths?